utils/validation/tests/validation_test.rs

```diff
@@ -0,0 +1,116 @@
+use std::{
+    env,
+    path::{Path, PathBuf},
+};
+
+use anyhow::bail;
+use clap::Clap;
+
+use casper_validation::{abi::ABIFixture, error::Error, test_case::TestCase, Fixture};
+
+fn get_fixtures_path() -> PathBuf {
+    let mut path = Path::new(env!("CARGO_MANIFEST_DIR")).to_path_buf();
+    path.push("tests");
+    path.push("fixtures");
+    path
+}
+
+fn prog() -> Option<String> {
+    let first_arg = env::args().next()?;
+    let path = Path::new(&first_arg);
+    let filename = path.file_name()?.to_str()?;
+    let prog_name = match filename.split('-').next() {
+        Some(name) => name,
+        None => filename,
+    };
+    Some(prog_name.to_string())
+}
+
+fn make_abi_tests(test_name: &str, test_fixture: ABIFixture) -> Vec<(String, Box<dyn TestCase>)> {
+    let prog_name = prog().expect("should get exe");
+
+    let mut tests = Vec::with_capacity(test_fixture.len());
+
+    for (test_case, data) in test_fixture.into_inner() {
+        // validation_test::fixture_file_name::test_case
+        let desc = format!("{}::{}::{}", prog_name, test_name, test_case);
+
+        tests.push((desc, Box::new(data) as Box<dyn TestCase>));
+    }
+
+    tests
+}
+
+type TestPair = (String, Box<dyn TestCase>);
+
+fn make_test_cases() -> Result<Vec<TestPair>, Error> {
+    let fixtures = get_fixtures_path();
+    let test_fixtures = casper_validation::load_fixtures(&fixtures)?;
+
+    let mut tests = Vec::new();
+
+    for test_fixture in test_fixtures {
+        match test_fixture {
+            Fixture::ABI {
+                name,
+                fixture: abi_test_case,
+            } => tests.append(&mut make_abi_tests(&name, abi_test_case)),
+        }
+    }
+
+    Ok(tests)
+}
+
+#[derive(Clap, Debug)]
+struct Opts {
+    #[clap(long = "nocapture", parse(from_flag = ::std::ops::Not::not))]
+    capture: bool,
+    #[allow(unused_attributes)]
+    #[clap(name = "FILTERS")]
+    filters: Vec<String>,
+}
```

Comment:
I don't think we need this (yet).

The `--nocapture` flag shouldn't stop the test runner from printing out the name and result of each test case, and the `filters` member is unused.

I'd recommend just removing this for now so we can get this PR merged, and we can then iterate on it over time to increase its functionality as required.

Reply:
Addressed in commit 7e1a3e11e19040ceeae81a40faff90f1790abfa2 where I just simply do non-zero exit code if at least one test case failed